### DISCUSSION TITLE
set GITHUB_REF from event?

### DISCUSSION THREAD

**Discussion by fauxbytes** [nektos/act#1453] [Q&A] [answered]
```
$ git br --show-current
bb
$ cat .github/workflows/bar.yml
name: bar
on: push
jobs:
  bar:
    runs-on: ubuntu-latest
    steps:
      - name: my-step
        run: |
          env | grep GITHUB_REF
$ cat push.json
{
  "push": {
    "ref": "refs/heads/development"
  }
}
$ act push -e push.json -j bar
[bar/bar] 🚀  Start image=catthehacker/ubuntu:act-latest
[bar/bar]   🐳  docker pull image=catthehacker/ubuntu:act-latest platform= username= forcePull=false
[bar/bar]   🐳  docker create image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[bar/bar]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[bar/bar] ⭐ Run Main my-step
[bar/bar]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/0] user= workdir=
| GITHUB_REF_TYPE=branch
| GITHUB_REF=refs/heads/bb
| GITHUB_REF_NAME=bb
[bar/bar]   ✅  Success - Main my-step
[bar/bar] 🏁  Job succeeded
```

Shouldn't `GITHUB_REF` be set as-per the event file? How do I get it to work as expected?
Thanks!

**Comment by KnisterPeter** [comment, marked as answer]
Act currently takes the ref from your local checked out repository. That means the ref always points to your local branch/commit.

**Reply by fauxbytes** [reply]
Thanks!
Isn't that a bug, though? [the docs](https://github.com/nektos/act#events) make no disclaimer for GITHUB_REF. It also explicitly affirms `head_ref` & `base_ref` can be overriden via an event. Is there a reasoning why a user could not expect `github.ref` to work similarly?

**Reply by KnisterPeter** [reply]
The main usecase for act is to allow executing actions locally to test them. And the default would then be write your workflow and use act to run it. You don't need to commit or anything like that. There is mostly not even an event file.

Feel free to raise a bug ticket or even better open a PR with your request.

**Comment by fauxbytes** [comment]
Hi. Wondering if there's anything required on my part wrt [the associated pending PR](https://github.com/nektos/act/pull/1537). Cheers!

**Comment by dannystaple** [comment]
Hmm, the value I am currently seeing doesn't match either the event file or the current checked out version. I've temporarily had to make an env override in the file-under-test with a GUTHUB_REF.
